src/projects/dualBackedToOne/MDualBackedToOne.sol

```diff
@@ -0,0 +1,294 @@
+        address secondaryBacker,
+        address admin,
+        address collateralManager,
+        address yieldRecipientManager,
+        address yieldRecipient
+    ) internal onlyInitializing {
+        if (yieldRecipientManager == address(0)) revert ZeroYieldRecipientManager();
+        if (admin == address(0)) revert ZeroAdmin();
+
+        __MExtension_init(name, symbol);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        _grantRole(COLLATERAL_MANAGER_ROLE, collateralManager);
+        _grantRole(YIELD_RECIPIENT_MANAGER_ROLE, yieldRecipientManager);
+
+        _setYieldRecipient(yieldRecipient);
+
+        MDualBackedToOneStorageStruct storage $ = _getMDualBackedToOneStorageLocation();
+        $.secondaryBacker = IERC20(secondaryBacker);
+        $.secondaryDecimals = IERC20(secondaryBacker).decimals();
+    }
+
+    /// @inheritdoc IERC20
+    function balanceOf(address account) public view override returns (uint256) {
+        return _getMDualBackedToOneStorageLocation().balanceOf[account];
+    }
+
+    /// @inheritdoc IERC20
+    function totalSupply() public view override returns (uint256) {
+        return _getMDualBackedToOneStorageLocation().totalSupply;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function secondaryBacker() public view returns (IERC20) {
+        return _getMDualBackedToOneStorageLocation().secondaryBacker;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function secondarySupply() public view returns (uint256) {
+        return _getMDualBackedToOneStorageLocation().secondarySupply;
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function yield() public view returns (uint256) {
+        unchecked {
+            uint256 balance_ = _mBalanceOf(address(this));
+            uint256 mBacking_ = totalSupply() - secondarySupply();
+
+            return balance_ > mBacking_ ? balance_ - mBacking_ : 0;
+        }
+    }
+
+    /// @inheritdoc IMDualBackedToOne
+    function claimYield() public returns (uint256) {
+        _beforeClaimYield();
+
+        uint256 yield_ = yield();
+
+        if (yield_ == 0) return 0;
+
+        emit YieldClaimed(yield_);
+
+        _mint(yieldRecipient(), yield_);
+
+        return yield_;
+    }
+
+    function setYieldRecipient(address account) external onlyRole(YIELD_RECIPIENT_MANAGER_ROLE) {
+        claimYield();
+
+        _setYieldRecipient(account);
+    }
+
+    function yieldRecipient() public view returns (address) {
+        return _getMDualBackedToOneStorageLocation().yieldRecipient;
+    }
+
+    function wrapSecondary(address recipient, uint256 amount) external onlySwapFacility {
+        // NOTE: `msg.sender` is always SwapFacility contract.
+        //       `ISwapFacility.msgSender()` is used to ensure that the original caller is passed to `_beforeWrap`.
+        _wrapSecondary(ISwapFacility(msg.sender).msgSender(), recipient, amount);
+    }
+
+    function replaceSecondary(uint256 amount) external onlyRole(COLLATERAL_MANAGER_ROLE) {
```

Comment:
discuss the best approach for this swap on team sync, consider doing it via swap facility too
note - account for rounding error for non-earner to earner (extension)

Reply:
Has been removed.